storage/operation/approvals.go

```diff
@@ -43,7 +43,7 @@ func RetrieveResultApproval(r storage.Reader, approvalID flow.Identifier, approv
 //     lockctx.Proof to prove the higher logic is holding the lock inserting the approval after checking
 //     that the approval is not already indexed.
 func UnsafeIndexResultApproval(lctx lockctx.Proof, w storage.Writer, resultID flow.Identifier, chunkIndex uint64, approvalID flow.Identifier) error {
-	if !lctx.HoldsLock(storage.LockMyResultApproval) {
+	if !lctx.HoldsLock(storage.LockIndexResultApproval) {
 		return fmt.Errorf("missing lock for index result approval for result: %v", resultID)
 	}
```

Comment:
Btw, I have mixed feeling about enforcing the lock here -- I think we should, 
1. it makes sense for me to enforce holding the lock, if we also check that we are not overwriting
2. But here, we are outsourcing the check to the caller, and overwrite nonetheless. At that point, I feel its not much gain over also outsourcing the entire responsibility about locking. Because the caller might just lock right before this call.

To be clear, I think we should do 1. the only place where this method is called is here: https://github.com/onflow/flow-go/blob/5df13a7439a55a426ae9ea08425eb280f92d4ff2/storage/store/approvals.go#L74-L101
and there we anyway check. So we might as well delegate the check entirely to the storage layer.